pymc/tests/distributions/test_mixture.py

```diff
@@ -329,13 +329,17 @@ def test_list_multivariate_components_deterministic_weights(self, weights, compo
 
     def test_component_choice_random(self):
         """Test that mixture choices change over evaluations"""
+        p_first_component = 0.5
+        n_draws = 200
         with Model() as m:
-            weights = [0.5, 0.5]
+            weights = [p_first_component, 1.0 - p_first_component]
             components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
             mix = Mixture.dist(weights, components)
-        draws = draw(mix, draws=10)
-        # Probability of coming from same component 10 times is 0.5**10
-        assert np.unique(draws > 0).size == 2
```

Comment:
It fails if all draws come from the same component because that's very unlikely to happen under the given binomial:

```python
stats.binomtest(0, 200, 0.5).pvalue  # 1.2446030555722283e-60
```

The assertion is stronger than the original one because it would also fail, say, 50 vs 150 draws from the components:
```python
stats.binomtest(50, 200, 0.5).pvalue # 8.393020875604775e-13
```

However, I fully agree, if the purpose of the test is to only check that both components exist in the draws, then the original version is simpler and to be preferred.